### PR TITLE
Security/avoid eval

### PR DESCRIPTION
**Pull request by fktt22**
Soundtrack of this PR: [Neptune, the mystic](https://youtu.be/Isic2Z2e2xs?t=2479)

### Motivation

The security rules at https://www.electronjs.org/docs/tutorial/security were checked. Only 15 is not satisfied (@davidyhall explained it was needed as is) but at least 16 makes it safer. The eval() avoidance rule was picked from a different website.

### In this PR

Using eval() is disallowed. 
Some function definitions with unused parameters were fixed.

### Caveats

_**Needs thorough manual check -- everything seems to work, but I cannot vouch for every function.**_
Some eslint rules had to be disabled because the original code didn't pass the pre-commit check.
A pair of tests do fail, but @davidyhall explained that it was expected, and should be fixed.

### Future Work

Review ESLint rules to avoid the described situation.
More security checks, if available.

### Screen Shots

None needed.

### Testing

Test Suites: 2 failed, 19 passed, 21 total
Tests:       75 passed, 75 total
Snapshots:   7 passed, 7 total
Time:        156.712 s
Ran all test suites.

### CODE REVIEW

File: app/main.dev.ts [hunk 1]
```diff
@@ -125,7 +131,9 @@ const createWindow = async () => {
           disableBlinkFeatures: 'Auxclick',
           enableRemoteModule: true,
           nodeIntegration: false,
+          nodeIntegrationInWorker: false,
           preload: path.join(__dirname, 'dist/renderer.prod.js'),
+          sandbox: true,
```
Comment (davidyhall) [inline app/main.dev.ts:136]: Are you able to run `sandbox: true` with `yarn build` or the package version in Linux? I get a black screen on a Mac.

**Review by davidyhall — COMMENTED** [review]
I'm having trouble with running this in `sandbox` mode for the distributed build or package. Does it work in Linux?

**Comment by davidyhall** [comment]
I jumped into your soundtrack and it started me off with Mars. Very intense. I reset the link to start at Neptune -- a very different vibe.

File: app/main.dev.ts (hunk 1 above)
Comment (fktt22) [reply app/main.dev.ts:136]: I'll try this out; I was using `yarn dev`.

Comment (fktt22) [reply app/main.dev.ts]: I can confirm `yarn dev` works (tested it again, to be sure) but the package built with `yarn package --linux tar.gz` failed. On further study, removing the option is wisest; done. I will include testing the built app (for Linux at least) in future code changes.

**Review by davidyhall — APPROVED** [review]
LGTM. -- I spent the night fighting the flaky form tests. I'm hoping the most recent merge lowers the flake probability to less than 10%. We'll see...

**Comment by davidyhall** [comment]
Retrying the test again, but heading to bed. Here's my comment about the flaky test from another PR: 

It's a fascinating bug. For some reason, my form tests may fail. If they do, the context is not defaulted to initialized, and the SplashScreen is rendered. Or, there's just an indefinite await that times out. I'm honestly not sure what's the issue. I'll consider breaking the custom render into different types (or maybe a more complex function that takes in options). Hopefully, by mounting a smaller render, it will better handle the problem. Until then, I'm going to try to expand the memory yet again to see if that unblocks the failure.

**Pull request merged**